### PR TITLE
feat(planner): support settings set sql_dialect

### LINKED ISSUE #7131: Feature: Add Setting `sql_dialect` to control the dialect of parser
**Summary**

Description for this feature.

Releated #7130  #7071

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support settings set `sql_dialect`

Fixes #7131

### CODE REVIEW

File: src/common/settings/src/lib.rs [hunk 1]
```diff
@@ -220,6 +239,17 @@ impl Settings {
...
+            },
+            SettingValue {
+                default_value: DataValue::String("PostgreSQL".as_bytes().to_vec()),
+                user_setting: UserSetting::create(
+                    "sql_dialect",
+                    DataValue::String("PostgreSQL".as_bytes().to_vec()),
+                ),
+                level: ScopeLevel::Session,
+                desc: "SQL dialect, support \"PostgreSQL\" and \"MySQL\", default value: \"PostgreSQL\"",
+                possible_values: Some(vec!["PostgreSQL", "MySQL"]),
```
Comment (sundy-li) [inline src/common/settings/src/lib.rs:252]: 👍

**Comment by sundy-li** [comment]
Need a test file in logictest.

For example:
```
set sql_dialect = 'xxxx';

select "33";
```

**Review by sundy-li — APPROVED** [review]

**Review by xudong963 — APPROVED** [review]

**Comment by leiysky** [comment]
We should provide a default SQL dialect for databend.

Before persisting any of the SQL query, the SQL text must be normalized into default dialect. Otherwise we will encounter bugs if user is creating a view with MySQL dialect but query with default dialect.

For example, user is creating a view with `create view v as select "A" from t` in MySQL dialect, then the token `"A"` is ambigious, which is impossible to be fixed.

**Comment by b41sh** [comment]
> @b41sh
> 
> We should provide a default SQL dialect for databend.
> 
> Before persisting any of the SQL query, the SQL text must be normalized into default dialect. Otherwise we will encounter bugs if user is creating a view with MySQL dialect but query with default dialect.
> 
> For example, user is creating a view with `create view v as select "A" from t` in MySQL dialect, then the token `"A"` is ambigious, which is impossible to be fixed.

the default SQL dialect is `PostgreSQL`, can we rewrite the view query to `PostgreSQL`?

**Comment by b41sh** [comment]
@leiysky the subquery of view is formatted as `PostgreSQL` dialect. When querying, whether it is `MySQL` or `PostgreSQL` dialect, we can get the correct results.

**Pull request merged**